src/kaleidoscope/layers.cpp

```diff
@@ -201,10 +186,16 @@ void Layer_::deactivate(uint8_t layer) {
   // Turn off its bit in layer_state_
   bitClear(layer_state_, layer);
 
-  // If the target layer was the previous highest active layer,
-  // update top_active_layer_
-  if (layer == top_active_layer_)
-    updateTopActiveLayer();
+  // Rearrange the activation order array...
+  uint8_t idx = 0;
+  for (uint8_t i = active_layer_count_; i > 0; i--) {
+    if (active_layers_[i] == layer) {
+      idx = i;
+      break;
+    }
+  }
+  memmove(&active_layers_[idx], &active_layers_[idx + 1], active_layer_count_ - idx);
+  active_layer_count_--;
```

Comment:
I may be missing something here, but it looks like there's no protection against deactivating the only layer remaining in the keymap.

Reply:
It's a sublte thing, but having now written a test explicitly for this case, I can now say with confidence, that there _is_ protection, just not in `deactivate()`: it's in `updateActiveLayers()`, which starts with `memset(active_layer_keymap_, 0, Runtime.device().numKeys());`. Why is this a protection? It is because `active_layer_keymap_` contains an address->layer mapping for each key on the device. If there are no active layers, we'll default to 0, just like we do prior to this PR.